Documentation/contributing.rst

```diff
@@ -227,15 +223,106 @@ This box will need to be updated when a new developer needs a new dependency
+
+See :ref:`dev_env` for information on how to setup the development environment.
+
+When the development VM is provisioned, it builds and installs Cilium.  After
+the initial build and install you can do further building and testing
+incrementally inside the VM. ``vagrant ssh`` takes you to the Cilium source
+tree directory (``/home/vagrant/go/src/github.com/cilium/cilium``) by default,
+and the following commands assume that you are working within that directory.
+
+Build Cilium
+^^^^^^^^^^^^
+
+Assuming you have synced (rsync) the source tree after you have made changes,
+or the tree is automatically in sync via NFS or guest additions folder sharing,
+you can issue a build as follows:
+
+::
+
+    $ make
+
+Install to dev environment
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After a successful build and test you can re-install Cilium by:
+
+::
+
+    $ sudo -E make install
+
+Restart Cilium service
+^^^^^^^^^^^^^^^^^^^^^^
+
+To run the newly installed version of Cilium, restart the service:
+
+::
+
+    $ sudo systemctl restart cilium
+
+You can verify the service and cilium-agent status by the following
+commands, respectively:
+
+::
+
+    $ sudo systemctl status cilium
+    $ cilium status
+
+Making Changes
+~~~~~~~~~~~~~~
+
+#. Create a topic branch: ``git checkout -b myBranch master``
+#. Make the changes you want
+#. Separate the changes into logical commits.
+
+   #. Describe the changes in the commit messages. Focus on answering the
+      question why the change is required and document anything that might be
+      unexpected.
+   #. If any description is required to understand your code changes, then
+      those instructions should be code comments instead of statements in the
+      commit description.
+#. Make sure your changes meet the following criteria:
+
+   #. New code is covered by :ref:`unit_testing`.
+   #. End to end integration / runtime tests have been extended or added. If
+      not required, mention in the commit message what existing test covers the
+      new code.
+   #. Follow-up commits are squashed together nicely. Commits should separate
+      logical chunks of code and not represent a chronological list of changes.
+#. Run ``git diff --check`` to catch obvious white space violations
+#. Run ``make`` to build your changes. This will also run ``go fmt`` and error out
+   on any golang formatting errors.
+#. See :ref:`unit_testing` on how to run unit tests.
+#. See :ref:`testsuite` for information how to run the end to end integration
+   tests
+
+.. _unit_testing:
+
 Unit Testing
-------------
+~~~~~~~~~~~~
 
 Cilium uses the standard `go test <https://golang.org/pkg/testing/>`__ framework
 in combination with `gocheck <http://labix.org/gocheck>`__ for richer testing
 functionality.
 
+Prerequisites
+^^^^^^^^^^^^^
+
+Some tests interact with the kvstore and depend on a local kvstore instances of
+both etcd and consul. To start the local instances, run:
+
+::
+
+     $ make start-kvstores
```

Comment:
Doesn't `make tests` run `make start-kvstores` automatically?

Reply:
This makes it clear in case somebody runs individual tests with `go test`

Reply:
Yes but will be not clear for someone that is about to run `make tests`.

Reply:
It should be a nop then and if not, we should fix it. It's a prerequisite for the entire chapter and only one make target takes care of it automatically.

Reply:
AFAICT it is a no-op if the kvstores are already started.